src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs

```diff
@@ -3,12 +3,199 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
+using EdFi.DmsConfigurationService.Backend.Repositories;
 using EdFi.DmsConfigurationService.DataModel;
+using EdFi.DmsConfigurationService.DataModel.Application;
+using EdFi.DmsConfigurationService.Frontend.AspNetCore.Infrastructure;
 using EdFi.DmsConfigurationService.Frontend.AspNetCore.Model.Validator;
+using FluentValidation;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
 
 namespace EdFi.DmsConfigurationService.Frontend.AspNetCore.Modules;
 
-public class ApplicationModule : BaseModule<Application, ApplicationValidator>
+public class ApplicationModule : IEndpointModule
 {
-    protected override string GetBaseRoute() => "/v2/applications";
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        endpoints.MapPost("/v2/applications/", InsertApplication).RequireAuthorizationWithPolicy();
+        endpoints.MapGet("/v2/applications/", GetAll).RequireAuthorizationWithPolicy();
+        endpoints.MapGet($"/v2/applications/{{id}}", GetById).RequireAuthorizationWithPolicy();
+        endpoints.MapPut($"/v2/applications/{{id}}", Update).RequireAuthorizationWithPolicy();
+        endpoints.MapDelete($"/v2/applications/{{id}}", Delete).RequireAuthorizationWithPolicy();
+    }
+
+    private static async Task<IResult> InsertApplication(
+        [FromServices] ApplicationInsertCommandValidator validator,
+        [FromBody] ApplicationInsertCommand command,
+        [FromServices] HttpContext httpContext,
+        [FromServices] IApplicationRepository applicationRepository,
+        [FromServices] IClientRepository clientRepository,
+        [FromServices] ILogger<ApplicationModule> logger
+    )
+    {
+        logger.LogDebug("Entering UpsertApplication");
+        await validator.GuardAsync(command);
+
+        Guid clientId = Guid.NewGuid();
+        string clientSecret = Guid.NewGuid().ToString();
+        if (
+            await clientRepository.CreateClientAsync(
+                clientId.ToString(),
+                clientSecret,
+                command.ApplicationName
+            )
+        )
```

Comment:
We may need explicitly handle the error from the Keycloak or backend services. Currently, system throws 500 internal server error.

Reply:
Johnny is working on ticket DMS-389 for this purpose. I left for him.